dataflow/main.py

```diff
@@ -20,88 +21,151 @@
+                | 'PeMS: Window' >> WindowInto(window)
         )
 
         weather: PCollection = (
                 pipeline
-                | "weather: Read" >> io.ReadFromBigQuery(table=get_tabel_spec("weather_partitioned", partition))
-                | "weather: Map" >> ParDo(WeatherTransformDoFn(segments))
-                | 'weather: Window' >> WindowInto(window)
+                | "Weather: Read" >> io.ReadFromBigQuery(
+            query=get_table_query(weather_table, start, end, window_size), use_standard_sql=True)
+                | "Weather: Map to Segments" >> ParDo(WeatherTransformDoFn(segments))
+                | 'Weather: Window' >> WindowInto(window)
         )
 
         result: PCollection = (({
             'bay_area_511_event': bay_area_511_event, 'weather': weather, 'pems': pems
         })
-                               | 'Merge' >> CoGroupByKey()
-                               | 'Encode' >> ParDo(SegmentFeatureTransformDoFn(segments, metadata_version)))
+                               | 'Merge by Segment' >> CoGroupByKey()
+                               | 'Feature Transform' >> ParDo(SegmentFeatureTransformDoFn(segments, metadata_version))
+                               | 'Discard Buffer' >> Filter(
+                    lambda row: start.timestamp() <= row['timestamp'] < end.timestamp()))
 
-        if save_to_bigquery:
+        if output_table is not None:
             (result
-             | 'bigquery: write' >> io.WriteToBigQuery(
-                        table=get_tabel_spec("processed_partitioned"),
+             | 'Write to BigQuery' >> io.WriteToBigQuery(
+                        table=get_tabel_spec(output_table),
                         create_disposition=BigQueryDisposition.CREATE_NEVER,
                         write_disposition=BigQueryDisposition.WRITE_APPEND))
 
         else:
             (result
-             | 'json: map' >> Map(json.dumps)
-             | 'json: write' >> WriteToText(f"{partition}_{metadata_version}.json"))
+             | 'Map to JSON' >> Map(json.dumps)
+             | 'Save as Local File' >> WriteToText(
+                        f"{start.strftime('%Y-%m-%d')}_{end.strftime('%Y-%m-%d')}_v{metadata_version}.json"))
 
 
 if __name__ == "__main__":
     logging.getLogger().setLevel(logging.WARNING)
 
     parser = argparse.ArgumentParser()
     parser.add_argument(
-        "--date",
-        help="Date (YYYYMMDD) to process within its 24-hour period in Pacific Standard Time (not Daylight Saving).",
+        "--start",
+        help="Date (YYYYMMDD) to start process at its 12:00AM in UTC Time (inclusive).",
         type=str,
         required=True
     )
     parser.add_argument(
-        "--metadata",
-        help="Version of the metadata to use.",
-        type=int,
+        "--end",
+        help="Date (YYYYMMDD) to end process at its 12:00AM in UTC Time (exclusive).",
+        type=str,
+        required=True
+    )
+    parser.add_argument(
```

Comment:
To me it looks like a pain to have to specify table name from the command line -consider default values?

Reply:
Just added the table names as default values. I designed it to come from the command line so users could have multiple simulations running without interfering each other.